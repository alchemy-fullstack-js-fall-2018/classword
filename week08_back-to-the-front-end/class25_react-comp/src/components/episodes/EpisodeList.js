import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import Pageable from '../pageable/Pageable';
import Episode from './Episode';
import { getEpisodes } from '../../services/rickAndMortyApi';

export default class EpisodeList extends Component {
  state = {
    currentPage: 1,
    totalPages: 10,
    episodes: []
  };

  fetchEpisode = () => {
    getEpisodes(this.state.currentPage) // Fetch episodes
      .then(({ pages, episodes }) => {
        // update episodes state
        // update totalPages
        this.setState({ episodes, totalPages: pages });
      });
  };

  componentDidMount() {
    this.fetchEpisode();
  }

  handlePageUpdate = page => {
    this.setState({ currentPage: page }, () => {
      this.fetchEpisode();
    });
  };

  render() {
    const { currentPage, totalPages, episodes } = this.state;
    const episodeComponents = episodes.map(episode => {
      return (
        <Episode key={episode.id}
          id={episode.id}
          title={episode.title}
          episodeNumber={episode.episodeNumber} />
      );
    });
    return (
      <Fragment>
        <Pageable currentPage={currentPage}
          totalPages={totalPages}
          updatePage={this.handlePageUpdate} />
        {episodeComponents}
      </Fragment>
    );
  }
}
