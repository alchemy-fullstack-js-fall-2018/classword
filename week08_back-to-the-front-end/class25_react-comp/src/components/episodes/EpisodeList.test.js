import React from 'react';
import { shallow } from 'enzyme';
import EpisodeList from './EpisodeList';

describe('EpisodeList', () => {
  it('renders an EpisodeList', () => {
    const wrapper = shallow(<EpisodeList />);
    expect(wrapper).toMatchSnapshot();
  });

  it('is a pageable list', () => {
    const wrapper = shallow(<EpisodeList />);
    expect(wrapper.find('Pageable')).toHaveLength(1);
  });
});
