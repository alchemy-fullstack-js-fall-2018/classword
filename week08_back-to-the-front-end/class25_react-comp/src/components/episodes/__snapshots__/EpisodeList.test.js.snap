// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeList renders an EpisodeList 1`] = `
<Fragment>
  <Pageable
    currentPage={1}
    totalPages={10}
    updatePage={[Function]}
  />
</Fragment>
`;
